README.md

```diff
@@ -1 +1,64 @@
-# saltoapis-dotnet
\ No newline at end of file
+# saltoapis-dotnet
+This repository contains the source code for the C# SALTO APIs SDK.
+
+> Some of this APIs are using gRPC. Refer to the [official gRPC documentation](https://docs.microsoft.com/en-us/aspnet/core/grpc/client?view=aspnetcore-5.0) to learn more about gRPC.
+
+
+## Authentication Example
+The SDK provides a simple gRPC interceptor that will automatically get and refresh valid access tokens and include them in all gRPC requests:
+```c#
+// create a gRPC channel
+GrpcChannel channel = GrpcChannel.ForAddress("https://nebula.saltoapis.com");
+
+// Create a new SaltoapisAuthInterceptor with your user credentials
+var authInterceptor = new SaltoapisAuthInterceptor(username, secret);
+
+// And add the interceptor to your channel
+var invoker = ChannelExtensions.Intercept(channel, authInterceptor);
+
+// Now you can use the resulting invoker to instantiate gRPC services
+var service = new UserService.UserServiceClient(invoker);
+```
+
+You can find more information about authentication on https://developer.saltonebula.com/api/authentication/.
+
+## Setup
+This SDK publishes NuGet packages in GitHub's Package Registry. You can see all published packages here: https://github.com/saltoapis/saltoapis-dotnet/packages/.
+
+Even though the packages are public, GitHub does not yet support downloading packages anonymously so you will need a GitHub access token with the `read:packages` permission set to retrieve the packages. 
+
+To setup your Nuget environment with your access token do the following:
+
+### 1. Create a personal access token
+Access https://github.com/settings/tokens and press on "Generate new token". Give the token the name and expiration you want and be sure to check the `read:packages` scope. Then press "Generate token".
+
+You will then have to copy your token (cause its going to be shown to you only once).
+
+### 2. Add the GitHub NuGet repository
+You have to add the saltoapis NuGet repository (https://nuget.pkg.github.com/saltoapis/index.json) in order to access the packages. You can do that manually editting your NuGet.Config file:
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    ...
+    <add key="saltoapis" value="https://nuget.pkg.github.com/saltoapis/index.json" />
+  </packageSources>
+
+  <packageSourceCredentials>
+    <saltoapis>
+        <add key="Username" value="YourGithubUsename" />
+        <add key="ClearTextPassword" value="YourToken" />
+    </saltoapis>
+  </packageSourceCredentials>
+
+</configuration>
+```
+
+Or using IDE. In Rider you can open the NuGet panel and in the "Sources" tab add the new sources you want. It will show you a modal where you will have to set the following:
+- Name: saltoapis
+- URL: https://nuget.pkg.github.com/saltoapis/index.json
+- User: YourGithubUser
+- Password: YourToken
```

Comment:
It might be helpful to follow the same pattern for these types of variables/placeholders in all the docs. i.e.

- User: {{YOUR_GITHUB_USER}}
- Pasword: {{YOUR_TOKEN}}

See https://developer.saltosystems.com/nebula/api/authentication/#generate-an-access-token and https://github.com/saltosystems/developer.saltosystems.com/pull/89 for more details.

Reply:
Also useful (this is roughly what we follow): https://developers.google.com/style/placeholders